pathd/path_pcep_cli.c

```diff
@@ -0,0 +1,1480 @@
+				 PCEP_VTYSH_ARG_DEAD_TIMER_MAX,
+				 group_opts->max_dead_timer_seconds);
+		lines++;
+	}
+	if (group_opts->pcep_request_time_seconds > 0) {
+		index += sprintf(buf + index, "  %s %d\n",
+				 PCEP_VTYSH_ARG_PCEP_REQUEST,
+				 group_opts->pcep_request_time_seconds);
+		lines++;
+	}
+	if (group_opts->state_timeout_inteval_seconds > 0) {
+		index += sprintf(buf + index, "  %s %d\n",
+				 PCEP_VTYSH_ARG_STATE_TIMEOUT,
+				 group_opts->state_timeout_inteval_seconds);
+		lines++;
+	}
+	if (group_opts->tcp_md5_auth[0] != '\0') {
+		index += sprintf(buf + index, "  %s %s\n",
+				 PCEP_VTYSH_ARG_TCP_MD5,
+				 group_opts->tcp_md5_auth);
+		lines++;
+	}
+	if (group_opts->draft07) {
+		index += sprintf(buf + index, "  %s\n",
+				 PCEP_VTYSH_ARG_SR_DRAFT07);
+		lines++;
+	}
+	if (group_opts->pce_initiated) {
+		index +=
+			sprintf(buf + index, "  %s\n", PCEP_VTYSH_ARG_PCE_INIT);
+		lines++;
+	}
+
+	return lines;
+}
+
+int pcep_cli_pcc_peer_config_write(struct vty *vty)
+{
+	int i = 0;
+	int lines = 0;
+	char buf[1024] = "";
+
+	for (; i < MAX_PCE; i++) {
+		struct pce_opts *pce_opts = pcep_g->pce_opts[i];
+		if (pce_opts == NULL) {
+			continue;
+		}
+
+		vty_out(vty, "pcc-peer %s\n", pce_opts->pce_name);
+		if (IS_IPADDR_V6(&pce_opts->addr)) {
+			vty_out(vty, "  %s %s %pI6", PCEP_VTYSH_ARG_ADDRESS,
+				PCEP_VTYSH_ARG_IPV6, &pce_opts->addr.ipaddr_v6);
+		} else {
+			vty_out(vty, "  address %s %pI4", PCEP_VTYSH_ARG_IP,
+				&pce_opts->addr.ipaddr_v4);
+		}
+		if (pce_opts->port != PCEP_DEFAULT_PORT) {
+			vty_out(vty, " %s %d", PCEP_VTYSH_ARG_PORT,
+				pce_opts->port);
+		}
+		vty_out(vty, "%s\n", buf);
+		lines += 2;
+
+		if (pce_opts->config_group_name[0] != '\0') {
+			vty_out(vty, "  config-group %s\n",
+				pce_opts->config_group_name);
+			lines++;
+		}
+
+		/* Only display the values configured on the PCE, not the values
+		 * from its optional pce-config-group, nor the default values */
+		lines += pcep_cli_print_config_group(&pce_opts->config_opts,
+						     buf);
+
+		vty_out(vty, "%s", buf);
+		buf[0] = '\0';
+	}
+
+	return lines;
+}
+
+int pcep_cli_pcep_config_group_write(struct vty *vty)
+{
+	int i = 0;
+	int lines = 0;
+	char buf[1024] = "";
+
+	for (; i < MAX_PCE; i++) {
+		struct pcep_config_group_opts *group_opts =
+			pcep_g->config_group_opts[i];
+		if (group_opts == NULL) {
+			continue;
+		}
+
+		vty_out(vty, "pce-config-group %s\n", group_opts->name);
+		lines += 1;
+
+		lines += pcep_cli_print_config_group(group_opts, buf);
+		vty_out(vty, "%s", buf);
+		buf[0] = 0;
+	}
+
+	return lines;
+}
+
+/*
+ * VTYSH command syntax definitions
+ */
+
+DEFUN(pcep_cli_debug, pcep_cli_debug_cmd,
+      "[no] debug pathd pcep [basic] [path] [message] [pceplib]",
+      NO_STR DEBUG_STR
+      "pathd debugging\n"
+      "pcep module debugging\n"
+      "module basic debugging\n"
+      "path structures debugging\n"
+      "pcep message debugging\n"
+      "pceplib debugging\n")
+{
+	return path_pcep_cli_debug(vty, argc, argv);
```

Comment:
I am not convinced that doing that is a good idea, these functions are not used from anywhere else and you loose the knowledge about want arguments are required and optional. In addition, all of these could use FRR provided utility functions that may not work outside because you cannot know what arguments are available in these functions. Moreover, these could/should be changed to DEFPY macros. What advantage is there in splitting this ?

Reply:
The advantage to splitting these is to keep all the syntax definitions together. I never liked the previous way of combining the syntax and the processing logic, as it made it difficult to see all of the different syntax commands. I am following this code, which is very clean: https://github.com/opensourcerouting/frr/blob/master/ldpd/ldp_vty_cmds.c#L32

Reply:
The code you are referring to is using DEFPY, and is NOT passing argc and argv to the functions. If you want to do it like this , it is fine, what I don't like is passing generic arguments to them and doing some parsing there.